sentence_transformers/SentenceTransformer.py

```diff
@@ -474,53 +493,51 @@ def save_to_hub(self,
             )
         full_model_name = repo_url[len(endpoint)+1:].strip("/")
 
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            # First create the repo (and clone its content if it's nonempty).
-            logger.info("Create repository and clone it if it exists")
-            repo = Repository(tmp_dir, clone_from=repo_url)
-
-            # If user provides local files, copy them.
-            if local_model_path:
-                copy_tree(local_model_path, tmp_dir)
-            else:  # Else, save model directly into local repo.
-                create_model_card = replace_model_card or not os.path.exists(os.path.join(tmp_dir, 'README.md'))
-                self.save(tmp_dir, model_name=full_model_name, create_model_card=create_model_card, train_datasets=train_datasets)
-
-            #Find files larger 5M and track with git-lfs
-            large_files = []
-            for root, dirs, files in os.walk(tmp_dir):
-                for filename in files:
-                    file_path = os.path.join(root, filename)
-                    rel_path = os.path.relpath(file_path, tmp_dir)
-
-                    if os.path.getsize(file_path) > (5 * 1024 * 1024):
-                        large_files.append(rel_path)
-
-            if len(large_files) > 0:
-                logger.info("Track files with git lfs: {}".format(", ".join(large_files)))
-                repo.lfs_track(large_files)
-
-            logger.info("Push model to the hub. This might take a while")
-            push_return = repo.push_to_hub(commit_message=commit_message)
-
-            def on_rm_error(func, path, exc_info):
-                # path contains the path of the file that couldn't be removed
-                # let's just assume that it's read-only and unlink it.
-                try:
-                    os.chmod(path, stat.S_IWRITE)
-                    os.unlink(path)
-                except:
-                    pass
+        repo_dir = local_repo_path
 
-            # Remove .git folder. On Windows, the .git folder might be read-only and cannot be deleted
-            # Hence, try to set write permissions on error
+        if local_repo_path is None:
+            repo_dir = tempfile.mkdtemp()
+
+        # First create the repo (and clone its content if it's nonempty).
+        logger.info("Create repository and clone it if it exists")
+        repo = Repository(repo_dir, clone_from=repo_url)
```

Comment:
Not really related to your PR, but could we use `create_repo` with `exist_ok` before `Repository`? That way we know that the repository exists on the Hub. The behavior of `Repository` creating the repo is a bit strange and will be deprecated in the future.

Reply:
Yup, it's already being called further up! https://github.com/UKPLab/sentence-transformers/blob/a83ce1629a1e6da8706ce8f6c5ce5c51ba06a947/sentence_transformers/SentenceTransformer.py#L486

I think the comment here is just confusing, so I'll change it.